crates/router/src/core/payment_methods/cards.rs

```diff
@@ -2861,35 +2915,99 @@ pub async fn filter_payment_methods(
+                                )));
+                            };
+                        });
+
+                    payment_attempt
                         .and_then(|attempt| attempt.mandate_data.as_ref())
                         .map(|mandate_detail| {
                             if mandate_detail.update_mandate_id.is_some() {
-                                filter_pm_based_on_update_mandate_support_for_connector(
-                                    supported_payment_methods_for_update_mandate,
-                                    &payment_method,
-                                    &payment_method_object.payment_method_type,
-                                    connector_variant,
-                                )
-                            } else {
-                                true
+                                context_values.push(dir::DirValue::PaymentType(
+                                    euclid::enums::PaymentType::UpdateMandate,
+                                ));
+                                if let Ok(connector) = api_enums::RoutableConnectors::from_str(
+                                    connector_variant.to_string().as_str(),
+                                ) {
+                                    context_values.push(dir::DirValue::Connector(Box::new(
+                                        api_models::routing::ast::ConnectorChoice {
+                                            connector,
+                                            #[cfg(not(feature = "connector_choice_mca_id"))]
+                                            sub_label: None,
+                                        },
+                                    )));
+                                };
                             }
+                        });
+
+                    payment_attempt
+                        .map(|attempt| {
+                            attempt.mandate_data.is_none() && attempt.mandate_details.is_none()
                         })
-                        .unwrap_or(true);
+                        .and_then(|res| {
+                            res.then(|| {
+                                context_values.push(dir::DirValue::PaymentType(
+                                    euclid::enums::PaymentType::NonMandate,
+                                ))
+                            })
+                        });
 
-                    let filter9 = req
+                    payment_attempt
+                        .and_then(|inner| inner.capture_method)
+                        .and_then(|capture_method| {
+                            (capture_method == common_enums::CaptureMethod::Manual).then(|| {
+                                context_values.push(dir::DirValue::CaptureMethod(
+                                    common_enums::CaptureMethod::Manual,
+                                ));
+                            })
+                        });
+
+                    let filter_incorrect_client_secret = req
```

Comment:
Can we rename it to something like `saved_payment_methods_filter` or something similar?